src/Microsoft.DotNet.Tools.Compiler/PPFileTokenizer.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Globalization;
+using System.Text;
+
+namespace Microsoft.DotNet.Tools.Compiler
+{
+    public class PPFileTokenizer
```

Comment:
The build task code is [here](https://github.com/NuGet/NuGet.BuildTasks/blob/b8552204ff93245bb6e36ff9a4cb6091d05a7b94/src/Microsoft.NuGet.Build.Tasks/Preprocessor.cs).